デプロイエラーの調査

CIで発覚している

```
Run bun run deploy
  bun run deploy
  shell: /usr/bin/bash -e {0}
  env:
    NEXT_PUBLIC_API_BASE_URL: ***
    CLOUDFLARE_ACCOUNT_ID: ***
    CLOUDFLARE_API_TOKEN: ***
$ opennextjs-cloudflare build && opennextjs-cloudflare deploy

┌─────────────────────────────┐
│ OpenNext — Cloudflare build │
└─────────────────────────────┘

App directory: /home/runner/work/effective-yomimono/effective-yomimono/frontend
Next.js version : 15.3.[2](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:2)
@opennextjs/cloudflare version: 1.0.2
@opennextjs/aws version: ^[3](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:3).6.1

┌─────────────────────────────────┐
│ OpenNext — Building Next.js app │
└─────────────────────────────────┘

$ next build
⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

   ▲ Next.js 15.3.2

   Creating an optimized production build ...
 ✓ Compiled successfully in 12.0s
   Linting and checking validity of types ...
   Collecting page data ...
   Generating static pages (0/8) ...
   Generating static pages (2/8) 
   Generating static pages ([4](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:4)/8) 
   Generating static pages (6/8) 
 ✓ Generating static pages (8/8)
   Finalizing page optimization ...
   Collecting build traces ...

Route (app)                                 Size  First Load JS
┌ ○ /                                    1.4[5](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:5) kB         115 kB
├ ○ /_not-found                            977 B         102 kB
├ ○ /favorites                             958 B         114 kB
├ ○ /labels                              4.[6](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:6)5 kB         115 kB
└ ○ /recent                                9[7](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:7)1 B         114 kB
+ First Load JS shared by all             101 kB
  ├ chunks/4bd1b696-db4fcbc743[8](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:9)5d0d0.js  53.2 kB
  ├ chunks/684-e32f13b326aeb8a6.js       45.[9](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:10) kB
  └ other shared chunks (total)          1.92 kB


○  (Static)  prerendered as static content


┌──────────────────────────────┐
│ OpenNext — Generating bundle │
└──────────────────────────────┘

Bundling middleware function...
Bundling static assets...
Bundling cache assets...
Building server function: default...
Applying code patches: 3.857s
# copyPackageTemplateFiles
⚙️ Bundling the OpenNext server...

Applying code patches:
 - patching require
 - patching cacheHandler
 - patching composableCache
 - patching 'require(this.middlewareManifestPath)'
 - patching `require.resolve` call
 - patching `require.resolve composable cache` call
All 6 patches applied

Worker saved in `.open-next/worker.js` 🚀

OpenNext build complete.

┌──────────────────────────────┐
│ OpenNext — Cloudflare deploy │
└──────────────────────────────┘


Populating KV incremental cache...
file:///home/runner/work/effective-yomimono/effective-yomimono/frontend/node_modules/@opennextjs/cloudflare/dist/cli/commands/populate-cache.js:89
        throw new Error(`No KV binding ${JSON.stringify(KV_CACHE_BINDING_NAME)} found!`);
              ^

Error: No KV binding "NEXT_INC_CACHE_KV" found!
    at populateKVIncrementalCache (file:///home/runner/work/effective-yomimono/effective-yomimono/frontend/node_modules/@opennextjs/cloudflare/dist/cli/commands/populate-cache.js:89:15)
    at populateCache (file:///home/runner/work/effective-yomimono/effective-yomimono/frontend/node_modules/@opennextjs/cloudflare/dist/cli/commands/populate-cache.js:[14](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:15)9:23)
    at async deploy (file:///home/runner/work/effective-yomimono/effective-yomimono/frontend/node_modules/@opennextjs/cloudflare/dist/cli/commands/deploy.js:4:5)
    at async file:///home/runner/work/effective-yomimono/effective-yomimono/frontend/node_modules/@opennextjs/cloudflare/dist/cli/index.js:43:1

Node.js v20.[19](https://github.com/ryosuke-horie/effective-yomimono/actions/runs/15082642205/job/42401271250#step:8:20).1
error: script "deploy" exited with code 1
Error: Process completed with exit code 1.
```

--- a/frontend/package.json
+++ b/frontend/package.json
@@ -9,7 +9,8 @@
 		"lint": "npx @biomejs/biome check",
 		"format": "npx @biomejs/biome check --write",
 		"preview": "opennextjs-cloudflare build && opennextjs-cloudflare preview",
-		"deploy": "opennextjs-cloudflare build && opennextjs-cloudflare deploy",
+		"deploy:setup": "bash ./setup-deploy.sh",
+		"deploy": "bun run deploy:setup && opennextjs-cloudflare build && opennextjs-cloudflare deploy",
 		"cf-typegen": "wrangler types --env-interface CloudflareEnv cloudflare-env.d.ts",
 		"knip": "knip"
 	},

--- a/frontend/setup-deploy.sh
+++ b/frontend/setup-deploy.sh
@@ -1,0 +1,44 @@
+#!/bin/bash
+
+# Check if required environment variables are set
+if [ -z "$CLOUDFLARE_API_TOKEN" ]; then
+  echo "CLOUDFLARE_API_TOKEN is not set"
+  exit 1
+fi
+
+if [ -z "$CLOUDFLARE_ACCOUNT_ID" ]; then
+  echo "CLOUDFLARE_ACCOUNT_ID is not set"
+  exit 1
+fi
+
+# Create KV namespace if it doesn't exist
+echo "Creating KV namespace for Next.js incremental cache..."
+KV_NAMESPACE_NAME="effective-yomimono-next-cache"
+
+# Check if namespace already exists
+EXISTING_NAMESPACE=$(wrangler kv:namespace list --compatibility-date=2024-09-26 | grep "\"$KV_NAMESPACE_NAME\"" | awk '{print $2}' | tr -d '"')
+
+if [ -z "$EXISTING_NAMESPACE" ]; then
+  # Create new namespace
+  echo "Creating new KV namespace: $KV_NAMESPACE_NAME"
+  KV_NAMESPACE_ID=$(wrangler kv:namespace create "$KV_NAMESPACE_NAME" --compatibility-date=2024-09-26 --preview false | grep "id =" | awk '{print $3}' | tr -d '"')
+else
+  # Use existing namespace
+  echo "Using existing KV namespace: $EXISTING_NAMESPACE"
+  KV_NAMESPACE_ID=$EXISTING_NAMESPACE
+fi
+
+if [ -z "$KV_NAMESPACE_ID" ]; then
+  echo "Failed to get KV namespace ID"
+  exit 1
+fi
+
+echo "KV namespace ID: $KV_NAMESPACE_ID"
+
+# Update wrangler.jsonc with the actual KV namespace ID
+sed -i.bak "s/PLACEHOLDER_KV_NAMESPACE_ID/$KV_NAMESPACE_ID/g" wrangler.jsonc
+
+# Remove backup file
+rm -f wrangler.jsonc.bak
+
+echo "Updated wrangler.jsonc with KV namespace ID"

--- a/frontend/wrangler.jsonc
+++ b/frontend/wrangler.jsonc
@@ -15,7 +15,13 @@
 	},
 	"observability": {
 		"enabled": true
-	}
+	},
+	"kv_namespaces": [
+		{
+			"binding": "NEXT_INC_CACHE_KV",
+			"id": "PLACEHOLDER_KV_NAMESPACE_ID"
+		}
+	]
 	/**
 	 * Smart Placement
 	 * Docs: https://developers.cloudflare.com/workers/configuration/smart-placement/#smart-placement

